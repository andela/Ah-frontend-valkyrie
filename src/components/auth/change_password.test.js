import React from "react";
import expect from "expect";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import ChangePassword from "./change_password";

const storeFake = (state, action) => ({
  subscribe: jest.fn(),
  default: jest.fn(),
  getState: () => state,
  dispatch: () => action,
});
const props = {
  changePasswordAction: jest.fn(),
  match: {
    params: {},
  },
};

const store = storeFake({
  passwordResetReducer:
    {
      user: {},
      message: "",
    },
});

describe("change password functionality", () => {
  const wrapper = mount(
    <Provider store={store}>
      <ChangePassword {...props} />
    </Provider>);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("handles state on submit", () => {
    const url = window.location;
    const arr = url.toString().split("/");
    const form = wrapper.find("#changePasswordForm");
    form.simulate("submit");
    expect(window.location.href).toEqual(`${arr[0]}//${arr[2]}/`);
  });

  it("handles state on change", () => {
    const event = { target: { name: "", value: "" } };
    const textField = wrapper.find(".form-control").first();
    const yoo = textField.simulate("change", event);
    expect(textField.instance().value).toEqual(event.target.value);
  });
});
