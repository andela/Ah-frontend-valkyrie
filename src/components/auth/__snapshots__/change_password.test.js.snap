// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change password functionality matches the snapshot 1`] = `ReactWrapper {}`;
